platform/qemu.go

```diff
@@ -364,3 +378,86 @@ func Virtio(board, device, args string) string {
 	}
 	return fmt.Sprintf("virtio-%s-%s,%s", device, suffix, args)
 }
+
+func setupIgnition(confPath string, diskImagePath string) error {
+	fileRemoteLocation := "/boot/ignition/config.ign"
+
+	// Set guestfish backend to direct in order to avoid libvirt as backend.
+	// Using libvirt can lead to permission denied issues if it does not have access
+	// rights to the qcow image
+	os.Setenv("LIBGUESTFS_BACKEND", "direct")
+	cmd := exec.Command("guestfish", "--listen", "-a", diskImagePath)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("getting stdout pipe: %v", err)
+	}
+
+	defer func() {
+		plog.Debugf("guestfish exit")
+		if err := exec.Command("guestfish", "--remote", "--", "exit").Run(); err != nil {
+			plog.Errorf("guestfish exit failed: %v", err)
+		}
+	}()
+	defer stdout.Close()
+
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("running guestfish: %v", err)
+	}
+	buf, err := ioutil.ReadAll(stdout)
+	if err != nil {
+		return fmt.Errorf("reading guestfish output: %v", err)
+	}
+	if err := cmd.Wait(); err != nil {
+		return fmt.Errorf("waiting for guestfish response: %v", err)
+	}
+	//GUESTFISH_PID=$PID; export GUESTFISH_PID
+	gfVarPid := strings.Split(string(buf), ";")
+	gfVarPidArr := strings.Split(gfVarPid[0], "=")
+	os.Setenv(gfVarPidArr[0], gfVarPidArr[1])
```

Comment:
This is missing a length check on both `gfVarPid` and `gfVarPidArr`. It will otherwise panic if parsing the file does not go as expected.